### ISSUE TITLE
Spec for JSON Decision Model (JDM) file

### ISSUE THREAD

**Issue by jerzy-wachowiak** [gorules/zen-go#42]
This is really impressive, what you have elaborated! 🎖🎖🎖

Can you please provide in your documentation any form of spec for [JSON Decision Model (JDM)](https://docs.gorules.io/reference/json-decision-model-jdm) files? (Have I overlooked something and the spec is in the documentation?)

The spec would help implement custom JSON Decision Model file creation...

**Comment by MTBorg** [comment]
+1, I haven't been able to find it either 😞

**Comment by stefan-gorules** [comment]
This is absolutely something we can add in. Would you be able to share a format that would make it easiest for you to understand the specification.

There will be a push to add some documentation and standardisation in future (in terms of having a formal document), though we aren't in a hurry.

For the moment we have the model "documented" as a:
- [Rust model](https://github.com/gorules/zen/blob/master/core/engine/src/model/mod.rs)
- [Zod schema](https://github.com/gorules/jdm-editor/blob/master/packages/jdm-editor/src/helpers/schema.ts)

Is one of the above something that you are looking for?

**Comment by MTBorg** [comment]
Yes, exactly, just some go-types that you can (de-)serialize to/fromt. I have an unfinished example here that you're free to use https://gist.github.com/MTBorg/5d8afc60413df789cd174dd642d02877.

In addition, having a versioned [jsonschema](https://json-schema.org/) would be a good quality improvement.
